src/App.tsx

```diff
@@ -1,26 +1,49 @@
-import React from 'react';
-import logo from './logo.svg';
+import React, { useState } from 'react';
+import TopElements from './components/topElements';
+import TotalPrice from './components/totalPrice';
+import ShoppingList from './components/shoppingList';
+import { IPurchase } from './utils/interfaces';
 import './App.css';
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.tsx</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+const purchases: IPurchase[] = [
+    {
+        place: 'shop1',
+        price: 10000,
+        date: '03.04.20',
+        isEdit: false,
+    },
+    {
+        place: 'shop2',
+        price: 12000,
+        date: '05.11.20',
+        isEdit: false,
+    },
+    {
+        place: 'shop3',
+        price: 8000,
+        date: '25.07.21',
+        isEdit: false,
+    },
+]
+
+const App = (): JSX.Element => {
+    const [purchase, setPurchase] = useState<IPurchase | undefined>()
+    const [shoppingList, setShoppingList] = useState<IPurchase[]>(purchases)
+
+    if (purchase) {
+        setShoppingList(() => [...shoppingList, purchase])
+        setPurchase(undefined)
+    }
```

Comment:
I don't think that this is correct. You should have a function called addPurchase - inside of it you implement the correct logic to add purchase. And you definitely don't need a callback inside of setShoppingList